src/Servers/Kestrel/samples/Http3SampleApp/Program.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+using Microsoft.Extensions.Logging;
+
+namespace Http3SampleApp
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var cert = CertificateLoader.LoadFromStoreCert("localhost", StoreName.My.ToString(), StoreLocation.CurrentUser, true);
+            var hostBuilder = new WebHostBuilder()
+                 .ConfigureLogging((_, factory) =>
+                 {
+                     factory.SetMinimumLevel(LogLevel.Trace);
+                     factory.AddConsole();
+                 })
+                 .UseKestrel()
+                 // TODO figure out how to make this fluent.
+                 // Things like APLN and cert should be able to be passed from corefx into bedrock
+                 .UseMsQuic(options =>
+                 {
+                     options.Certificate = cert;
+                     options.RegistrationName = "Quic";
+                     options.Alpn = "h3-24";
+                     options.IdleTimeout = TimeSpan.FromHours(1);
+                 })
+                 .ConfigureKestrel((context, options) =>
+                 {
+                     var basePort = 5555;
+
+                     options.Listen(IPAddress.Any, basePort, listenOptions =>
+                     {
+                         listenOptions.UseHttps();
```

Comment:
Is there some way to flow the UseHttps options down to the MsQuic layer? Having to configure this in two places is confusing.
Why do you even need UseHttps here at the moment since you're configuring MsQuic separately?

Reply:
Its not needed for http/3 I assume.

Reply:
Does MsQuic allow configuring TLS on a per-connection basis, or does all the configuration happen before binding? If it's the latter, I don't see how we could ever make the UseHttps() api work with MsQuic.

Reply:
All the config is setup at the beginning, it doesn't change per connection. The ask would be to pass the HttpsConnectionAdapterOptions down to MsQuic when initializing the transport. We want enabling H3 to be as transparent as possible.

Reply:
I don't think I'm going to touch anything now, but there are actually a few things we need to decide here:
- How to propagate TLS settings with/without calling UseHttps.
- MsQuicConnection still goes through the HttpsMiddleware. As brought up in our MultiplexedFactory discussion, maybe there should be no middleware pipeline for the connection?
- The MsQuicStream sets a fake TLS middleware, which we should check. I'll do that change now. I'll also add it to MsQuicConnection for now.